test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { ERC20ForwarderFactory } from "contracts/src/token/ERC20ForwarderFactory.sol";
+import { HyperdriveFactory } from "contracts/src/factory/HyperdriveFactory.sol";
+import { IERC20 } from "contracts/src/interfaces/IERC20.sol";
+import { IHyperdrive } from "contracts/src/interfaces/IHyperdrive.sol";
+import { AssetId } from "contracts/src/libraries/AssetId.sol";
+import { ETH } from "contracts/src/libraries/Constants.sol";
+import { FixedPointMath, ONE } from "contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveTest } from "test/utils/HyperdriveTest.sol";
+import { HyperdriveUtils } from "test/utils/HyperdriveUtils.sol";
+import { Lib } from "test/utils/Lib.sol";
+
+/// @author DELV
+/// @title IntegrationTest
+/// @notice The base contract for the integration testing suite.
+/// @dev A testing suite that provides a foundation to setup, deploy, and
+/// test common cases for Hyperdrive instances.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract IntegrationTest is HyperdriveTest {
```

Comment:
It would probably be better to call this `InstanceTest` to disambiguate it from the integration tests in `test/integration`.